### ISSUE TITLE
RDLibrary: Missing end-mark on some cuts.

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#808]
Hi.

Athough VERY RARE, I'm finding that in certain cuts, the marker editor does not show the end-mark.
I'd like to remark that it is very rare (I've seen this in a few cuts out of almost a thousand carts)... but it's there.
My guess is that it is just some front-end/qt5 issue whenever a cut has some length that prevents the GUI components/arangement to be rendered somehow, since the actual End-Mark time is sown in the table as read from backend/DB.

Some screenshots.

![Screenshot at 2022-05-17 11-38-46](https://user-images.githubusercontent.com/1419896/168989394-d1a71a57-dfc4-489a-8a95-433b32715d62.png)

![Screenshot at 2022-05-10 18-10-44](https://user-images.githubusercontent.com/1419896/168989415-fe2bb41e-6ee0-4525-9adb-5e87eead49f4.png)

Cheers!

**Comment by CWJR** [comment]
I can confirm that this has happened (but rarely) back to v2.10.2.

--Chuck

**Comment by alexolivan** [comment]
I think I've found some 'logic' or kind of pattern to replicate the problem.

Since I've been lately creating shorter 'radio edit' cuts on my carts, in order to prioritize the longer carts to work with first, I sorted carts by duration on RDLibrary, longer first, and started my work one by one down the list, in my spare time... Well, as I approached the 06:30 duration boundary, I realized that every cut I check, has end-mark outside the working area:

- All carts above (longer duration) my 06:33.0 cart/cut, show End-Mark normally.
- With 06:33.0 duration, I find the first cart/cut with End-Mark invisible.
- Following 6 cart/cuts, with durations between 06:33.0 and 06:24.5, all have missing End-Mark.
- With 06:24.5 duration, a cart/cut is the last one with End-Mark invisible
- With 06:24.3 duration, next cart/cut End-Mark is again visible, in a workspace that seems to double actual waveform length
- All carts bellow (shorter duration) do show End-Mark again

So, something seems to happen around 06:30... I wonder whether the same issue would repeat for a 13:00 minute cart/cut.
Hope those somehow acurate boundaries help pinpointing the source of the problem.

Regards.
